File: clients/ruby/lib/order_api.rb
require "uri"

class Order_api
  basePath = "https://www.bitmex.com/api/v1"
  # apiInvoker = APIInvoker

  def self.escapeString(string)
    URI.encode(string.to_s)
  end

  def self.get_orders (symbol,filter,columns,start,reverse,start_time,end_time,count= 100,opts={})
    query_param_keys = [:symbol,:filter,:columns,:count,:start,:reverse,:start_time,:end_time]

    # set default values and merge with input
    options = {
    :symbol => symbol,
      :filter => filter,
      :columns => columns,
      :start => start,
      :reverse => reverse,
      :start_time => start_time,
      :end_time => end_time,
      :count => count}.merge(opts)

    #resource path
    path = "/order".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = nil
    post_body = nil
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>headers, :body=>post_body }).make.body
    response.map {|response|Order.new(response)}

  end

def self.new_order (symbol,quantity,price,ioc,cl_ord_i_d,opts={})
    query_param_keys = []

    # verify existence of params
    raise "symbol is required" if symbol.nil?
    raise "quantity is required" if quantity.nil?
    raise "price is required" if price.nil?
    # set default values and merge with input
    options = {
    :symbol => symbol,
      :quantity => quantity,
      :price => price,
      :ioc => ioc,
      :cl_ord_i_d => cl_ord_i_d}.merge(opts)

    #resource path
    path = "/order".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = nil
    post_body = nil
    response = Swagger::Request.new(:POST, path, {:params=>queryopts,:headers=>headers, :body=>post_body }).make.body
    Order.new(response)

  end

def self.cancel_order (order_i_d,cl_ord_i_d,text,opts={})
    query_param_keys = []

    # set default values and merge with input
    options = {
    :order_i_d => order_i_d,
      :cl_ord_i_d => cl_ord_i_d,
      :text => text}.merge(opts)

    #resource path
    path = "/order".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = nil
    post_body = nil
    response = Swagger::Request.new(:DELETE, path, {:params=>queryopts,:headers=>headers, :body=>post_body }).make.body
    response.map {|response|Order.new(response)}

  end

def self.cancel_all (symbol,text,opts={})
    query_param_keys = []

    # set default values and merge with input
    options = {
    :symbol => symbol,
      :text => text}.merge(opts)

    #resource path
    path = "/order/all".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = nil
    post_body = nil
    response = Swagger::Request.new(:DELETE, path, {:params=>queryopts,:headers=>headers, :body=>post_body }).make.body
    object.new(response)

  end

def self.cancel_all_after (timeout,opts={})
    query_param_keys = []

    # verify existence of params
    raise "timeout is required" if timeout.nil?
    # set default values and merge with input
    options = {
    :timeout => timeout}.merge(opts)

    #resource path
    path = "/order/cancelAllAfter".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = nil
    post_body = nil
    response = Swagger::Request.new(:POST, path, {:params=>queryopts,:headers=>headers, :body=>post_body }).make.body
    object.new(response)

  end

end

File: clients/java/android/src/main/java/com/wordnik/client/api/OrderApi.java
package com.wordnik.client.api;

import com.wordnik.client.ApiException;
import com.wordnik.client.ApiInvoker;
import com.wordnik.client.model.Error;
import com.wordnik.client.model.Object;
import com.wordnik.client.model.Order;
import java.util.*;
import java.io.File;

public class OrderApi {
  String basePath = "https://www.bitmex.com/api/v1";
  ApiInvoker apiInvoker = ApiInvoker.getInstance();

  public void addHeader(String key, String value) {
    getInvoker().addDefaultHeader(key, value);
  }

  public ApiInvoker getInvoker() {
    return apiInvoker;
  }

  public void setBasePath(String basePath) {
    this.basePath = basePath;
  }

  public String getBasePath() {
    return basePath;
  }

  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public Order newOrder (String symbol, Double quantity, Double price, Boolean ioc, String clOrdID) throws ApiException {
    // verify required params are set
    if(symbol == null || quantity == null || price == null ) {
       throw new ApiException(400, "missing required params");
    }
    // create path and map variables
    String path = "/order/new".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, null, headerParams, contentType);
      if(response != null){
        return (Order) ApiInvoker.deserialize(response, "", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public Order newOrder_OrderApi_0 (String symbol, Double quantity, Double price, Boolean ioc, String clOrdID) throws ApiException {
    // verify required params are set
    if(symbol == null || quantity == null || price == null ) {
       throw new ApiException(400, "missing required params");
    }
    // create path and map variables
    String path = "/order".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, null, headerParams, contentType);
      if(response != null){
        return (Order) ApiInvoker.deserialize(response, "", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public List<Order> cancelOrder (String orderID, String clOrdID, String text) throws ApiException {
    // create path and map variables
    String path = "/order".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "DELETE", queryParams, null, headerParams, contentType);
      if(response != null){
        return (List<Order>) ApiInvoker.deserialize(response, "List", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public List<Order> getOrders (Object filter, List<any> columns, Double count) throws ApiException {
    // create path and map variables
    String path = "/order".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(filter)))
      queryParams.put("filter", String.valueOf(filter));
    if(!"null".equals(String.valueOf(columns)))
      queryParams.put("columns", String.valueOf(columns));
    if(!"null".equals(String.valueOf(count)))
      queryParams.put("count", String.valueOf(count));
    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams, contentType);
      if(response != null){
        return (List<Order>) ApiInvoker.deserialize(response, "List", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public List<Order> cancelOrder_OrderApi_0 (String orderID, String clOrdID, String text) throws ApiException {
    // verify required params are set
    if(orderID == null ) {
       throw new ApiException(400, "missing required params");
    }
    // create path and map variables
    String path = "/order/cancel".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, null, headerParams, contentType);
      if(response != null){
        return (List<Order>) ApiInvoker.deserialize(response, "List", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public List<Order> getOrders_OrderApi_0 (Object filter, List<any> columns, Double count) throws ApiException {
    // create path and map variables
    String path = "/order/myOrders".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(filter)))
      queryParams.put("filter", String.valueOf(filter));
    if(!"null".equals(String.valueOf(columns)))
      queryParams.put("columns", String.valueOf(columns));
    if(!"null".equals(String.valueOf(count)))
      queryParams.put("count", String.valueOf(count));
    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams, contentType);
      if(response != null){
        return (List<Order>) ApiInvoker.deserialize(response, "List", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public List<Order> getOrders_OrderApi_1 (Object filter, List<any> columns, Double count) throws ApiException {
    // create path and map variables
    String path = "/order/myOpenOrders".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(filter)))
      queryParams.put("filter", String.valueOf(filter));
    if(!"null".equals(String.valueOf(columns)))
      queryParams.put("columns", String.valueOf(columns));
    if(!"null".equals(String.valueOf(count)))
      queryParams.put("count", String.valueOf(count));
    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams, contentType);
      if(response != null){
        return (List<Order>) ApiInvoker.deserialize(response, "List", Order.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  //error info- code: 200 reason: "Request was successful" model: <none>
  //error info- code: 400 reason: "Parameter Error" model: Error
  //error info- code: 401 reason: "Unauthorized" model: Error
  //error info- code: 404 reason: "Not Found" model: Error
  public Object cancelAllAfter (Double timeout) throws ApiException {
    // verify required params are set
    if(timeout == null ) {
       throw new ApiException(400, "missing required params");
    }
    // create path and map variables
    String path = "/order/cancelAllAfter".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    String contentType = "application/json";

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, null, headerParams, contentType);
      if(response != null){
        return (Object) ApiInvoker.deserialize(response, "", Object.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
        return null;
      }
      else {
        throw ex;
      }
    }
  }
  }

